teuthology/provision/openstack.py

```diff
@@ -93,7 +93,15 @@ def attach_volumes(self, name, volumes):
                                      " not information available yet")
             # do not use OpenStack().run because its
             # bugous for volume
-            misc.sh("openstack server add volume " + name + " " + volume_name)
+            with safe_while(sleep=20, increment=20, tries=10,
+                            action="add volume " + volume_name) as proceed:
+                while proceed():
+                    try:
+                        misc.sh("openstack server add volume " + name + " " + volume_name)
+                        # misc.sh raises CalledProcessError if exit status != 0
```

Comment:
Do we need this comment here? It is better to this comment to the pydoc for the 'sh' in 'misc.py' if you need the clarity, and docs update is probably outside of this patch.